scheduler/replica_scheduler.go

```diff
@@ -691,6 +709,236 @@ func (rcs *ReplicaScheduler) getDiskWithMostUsableStorage(disks map[string]*Disk
+			bestDisk = disk
+		}
+	}
+
+	if bestDisk == nil || bestDiskScore == math.MaxFloat64 {
+		return nil, fmt.Errorf("no candidate disk found for node %v", nodeID)
+	}
+
+	return bestDisk, nil
+}
+
+// computeHybridBalanceScore blends absolute imbalance and ratio imbalance.
+func computeHybridBalanceScore(
+	usableStorage map[string]int64,
+	totalStorage map[string]int64,
+	alpha float64,
+) float64 {
+
+	if alpha >= 1 {
+		// Only absolute imbalance is considered.
+		return computeBalanceScoreFromUsableStorage(usableStorage)
+	}
+
+	if alpha <= 0 {
+		// Only ratio imbalance is considered.
+		return computeBalanceScoreFromUsableStorageRatio(usableStorage, totalStorage)
+	}
+
+	// 0 < alpha < 1
+	// Both absolute and ratio imbalance are considered.
+	// Formula: hybrid = alpha*ratioImbalance + (1-alpha)*absoluteImbalance
+	absBalanceScore := computeBalanceScoreFromUsableStorage(usableStorage)
+	ratioBalanceScore := computeBalanceScoreFromUsableStorageRatio(usableStorage, totalStorage)
+
+	if absBalanceScore == math.MaxFloat64 || ratioBalanceScore == math.MaxFloat64 {
+		// If either absolute or ratio imbalance cannot be computed (e.g.:
+		//   - no entries to compare,
+		//   - total capacity <= 0 making ratios invalid,
+		//   - or all usable values are 0 meaning everything is full),
+		// then return MaxFloat64 to signal "worst possible balance".
+		// This effectively prevents the scheduler from selecting this node/disk.
+		return math.MaxFloat64
+	}
+
+	return alpha*absBalanceScore + (1.0-alpha)*ratioBalanceScore
+}
+
+// computeBalanceScoreFromUsableStorage calculates imbalance using absolute usable bytes.
+func computeBalanceScoreFromUsableStorage(usableStorage map[string]int64) float64 {
+	// Formula: (max(usable) - min(usable)) / mean(usable)
+	sum := int64(0)
+	maxBytes := int64(math.MinInt64)
+	minBytes := int64(math.MaxInt64)
+
+	// just in case to prevent division by zero, should not happen
+	if len(usableStorage) == 0 {
+		return math.MaxFloat64
+	}
+
+	for _, usableBytes := range usableStorage {
+		sum += usableBytes
+		if usableBytes > maxBytes {
+			maxBytes = usableBytes
+		}
+		if usableBytes < minBytes {
+			minBytes = usableBytes
+		}
+	}
+
+	mean := float64(sum) / float64(len(usableStorage))
+	if mean == 0 {
+		// if all usable are zero, imbalance is undefined (everything is full).
+		// Safer to return math.MaxFloat64 so the scheduler avoids placing there
+		return math.MaxFloat64
+	}
+
+	return float64(maxBytes-minBytes) / mean
+}
+
+// computeBalanceScoreFromUsableStorageRatio calculates imbalance using usable ratios.
+func computeBalanceScoreFromUsableStorageRatio(usableStorage map[string]int64, totalStorage map[string]int64) float64 {
+	// Ratio = usable / total for each entry.
+	// Formula: (max(ratio) - min(ratio)) / mean(ratio)
+
+	if len(usableStorage) == 0 || len(totalStorage) == 0 {
+		return math.MaxFloat64
+	}
+
+	sumRatio := 0.0
+	maxRatio := -math.MaxFloat64
+	minRatio := math.MaxFloat64
```

Comment:
Can change to IEEE754 `Inf` so that we don't need to worry about overflow, and is easy to check with `math.IsInf`.
```suggestion
	maxRatio := math.Inf(-1)
	minRatio := math.Inf(1)
```

Reply:
Thanks, it's better